### PR TITLE
Resolve breaking changes in Boost 1.67

### PR DESCRIPTION
**Pull request by devinus**

### CODE REVIEW

File: rai/lib/numbers.cpp [hunk 1]
```diff
@@ -443,13 +442,13 @@ rai::uint512_union rai::sign_message (rai::raw_key const & private_key, rai::pub
...
 
-void rai::deterministic_key (rai::uint256_union const & seed_a, uint32_t index_a, rai::uint256_union & prv_a)
+void rai::deterministic_key (rai::uint256_union const & seed_a, uint64_t index_a, rai::uint256_union & prv_a)
 {
 	blake2b_state hash;
 	blake2b_init (&hash, prv_a.bytes.size ());
 	blake2b_update (&hash, seed_a.bytes.data (), seed_a.bytes.size ());
 	rai::uint256_union index (index_a);
-	blake2b_update (&hash, reinterpret_cast<uint8_t *> (&index.dwords[7]), sizeof (uint32_t));
+	blake2b_update (&hash, reinterpret_cast<uint8_t *> (&index.dwords[7]), sizeof (index.dwords[7]));
```
Comment (PlasmaPower) [inline rai/lib/numbers.cpp]: Why convert index to a `rai::uint256_t`? Are we changing the endianness? If so, I think we should use boost's functions explicitly for that.

File: rai/node/wallet.cpp [hunk 2]
```diff
@@ -465,7 +465,7 @@ rai::key_type rai::wallet_store::key_type (rai::wallet_value const & value_a)
 	}
 	else
 	{
-		if ((number >> 32).convert_to<uint32_t> () == 1)
+		if ((number >> 64).convert_to<uint64_t> () == 1)
```
Comment (PlasmaPower) [inline rai/node/wallet.cpp]: Are you sure that this change is correct? I think this would require a database upgrade.

**Comment by PlasmaPower** [comment]
Also, there are a couple of small formatting problems causing the Travis error.

File: rai/node/wallet.cpp (hunk 2 above)
Comment (devinus) [reply rai/node/wallet.cpp]: @PlasmaPower No, the PR is a bit premature. I wanted to run Travis tests against it.

**Comment by clemahieu** [comment]
What about boost 1.67 is causing these changes to be needed?  Some types can’t be widened because they’re part of the current wire protocol like the frontier_req 32bit types.

**Comment by devinus** [comment]
@clemahieu Sorry! The PR was premature and included some things I just wanted tested on Travis, I had hoped nobody would see it before I whittled it down to just the changes actually needed. 😳

**Comment by devinus** [comment]
@PlasmaPower @clemahieu Ready for review.

**Comment by PlasmaPower** [comment]
LGTM

**Pull request merged**